container/go/pkg/oci/image.go

```diff
@@ -0,0 +1,192 @@
+/// Copyright 2015 The Bazel Authors. All rights reserved.
```

Comment:
Do we have any level of unit testing for this code (and other bits of go code you are modifying)? Should we (maybe in a different PR given the potential effort / size)?

Reply:
This code is end to end tested. We spin up a local docker registry and push images in OCI format to that registry. The docker registry is in the best position to validate an image. However, I have an upcoming change that is going to fix a performance issue in the current Go code & functionality gaps which I plan to add more tests for. I might make those a combination of `container_test` for *unit-test* style testing & end to end tests for more thorough docker image schema validation. We can discuss more in that PR (will be after one more PR where I add a `container_flatten` implementation in Go).

But in general yes, I acknowledge this comment and in general, I plan to add some tests before the big switch.